datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1758,7 +1758,7 @@ impl<S: SimplifyInfo> TreeNodeRewriter for Simplifier<'_, S> {
             // try_cast/cast(expr as data_type) op literal
             Expr::BinaryExpr(BinaryExpr { left, op, right })
                 if is_cast_expr_and_support_unwrap_cast_in_comparison_for_binary(
-                    info, &left, &right,
+                    info, &left, &right, op,
```

Comment:
add `op` between left and right args -- this helps understand the role of the two `&Expr` args to the function
(or maybe even pass the whole `&BinaryExpr` into the function)